lib/inspec/profile.rb

```diff
@@ -316,6 +316,25 @@ def check # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metri
       m_unsupported.each { |u| warn.call(meta_path, 0, 0, nil, "doesn't support: #{u}") }
       @logger.info 'Metadata OK.' if m_errors.empty? && m_unsupported.empty?
 
+      # verify that a lockfile is present if we have dependencies
```

Comment:
I like the implementation, but I think it is in the wrong class.  We have a class for the lockfile, and one for the metadata.  Why is it Profile's job to validate them?  I could see `metadata.check_for_stale_lockfile(lockfile)` or `lockfile.is_current?(metadata)` or similar.  

It's a matter of opinion; but Profile does too many things already, I think.

Reply:
I was thinking about this as well. I was thinking that we may need to externalize all checks instead of keeping them in profiles. I am not sure that we should put the check logic into lockfile or metadata either since checks should be completely independent. Maybe checks should be an inspec profile that checks inspec profiles?

![](https://media.giphy.com/media/7pHTiZYbAoq40/giphy.gif)